hydroflow/src/util/mod.rs

```diff
@@ -250,3 +250,51 @@ mod test {
         );
     }
 }
+
+use std::io::Read;
+use std::process::{Child, ChildStdin, ChildStdout, Stdio};
+
+/// TODO:
+pub fn wait_for_output(output_so_far: &mut String, output: &mut ChildStdout, wait_for: &str) {
+    let re = regex::Regex::new(wait_for).unwrap();
+
+    while !re.is_match(output_so_far) {
+        println!("waiting: {}", output_so_far);
+        let mut buffer = [0u8; 1024];
+        let bytes_read = output.read(&mut buffer).unwrap();
+
+        if bytes_read == 0 {
+            panic!();
+        }
+
+        output_so_far.push_str(&String::from_utf8_lossy(&buffer[0..bytes_read]));
+    }
+}
+
+/// TODO:
+pub fn spawn(test_name: &str, args: &str) -> (Child, ChildStdin, ChildStdout) {
+    let mut server = if args.is_empty() {
+        std::process::Command::new("cargo")
+            .args(["run", "-p", "hydroflow", "--example"])
+            .arg(test_name)
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .spawn()
+            .unwrap()
+    } else {
+        std::process::Command::new("cargo")
+            .args(["run", "-p", "hydroflow", "--example"])
+            .arg(test_name)
+            .arg("--")
+            .args(args.split(' '))
```

Comment:
Can we make args of type `args: impl IntoIterator<Item = &str>` and make this the callers responsibility? Just for code injection sake :P

Reply:
it would mean the callers now have to use str::split(' ') or they use array literals and both of those add a lot of text to the many places it's called. The current usage mirrors calling it from the command line where you just pass all the arguments as one big string.